### PR TITLE
test: a test framework for verifying upgrade of Zoe and ZCF

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
refs: #6678

## Description

A test framework for verifying upgrade of Zoe and ZCF. This includes bootstrap scripts that can be used to upgrade Zoe and ZCF.

The test starts from the currently installed version of Zoe and ZCF. First it verifies that reallocation via staging and via the helper work and that the version internal to ZCF is not present. It then upgrades Zoe and ZCF as necessary to introduce the new behavior. Finally, it re-runs the initial verification to show that the ZCF internal version works.

Variants of this test will run [on top of](https://github.com/Agoric/agoric-sdk/pull/7966) the PR that [updates Zoe and Zcf](https://github.com/Agoric/agoric-sdk/pull/7946) to verify that upgrading to the new Zoe/ZCF works, and in
deployment/upgrade-test to show that the upgrade to the new Zoe & Zcf continues to succeed. 

### Security Considerations

Validates the upgrade plan

### Scaling Considerations

None

### Documentation Considerations

None

### Testing Considerations

It's all about the testing.

### CODE REVIEW

**Comment by Chris-Hibbert** [comment]
@turadg @dckc  This test isn't intended to be merged. #7966 contains all the relevant changes that should become part of the source tree. The point of this PR is to demonstrate that #7966 would succeed in a repo that didn't already contain the changes to Zoe and ZCF.

So: FYI, but not intended to be approved or merged.

File: packages/vats/test/bootstrapTests/supports.js [hunk 1]
```diff
@@ -488,4 +498,96 @@ export const makeSwingsetTestKit = async (
...
+    storage,
+    agoricNamesRemotes,
+  );
+  console.timeLog('DefaultTestContext', 'walletFactoryDriver');
+
+  console.timeEnd('DefaultTestContext');
+
+  const buildProposal = makeProposalExtractor({
+    childProcess: processAmbient,
+    fs: fsAmbient.promises,
```
Comment (dckc) [inline packages/vats/test/bootstrapTests/supports.js:539]: > modules should not export powerful objects (for example, objects that close over fs or process.env)
 -- [OCap discipline](https://github.com/Agoric/agoric-sdk/wiki/OCap-Discipline)

Please let's pass powers such as `fs` and `childProcess` into `makeTestContext` explicitly, if we're going to export it.

(yes, it's "just" test code, but I don't think this package as an `exports` thing that prevents unwanted deep imports, so this is actually part of the public API).

**Comment by Chris-Hibbert** [comment]
Converting to **DRAFT** to indicate that this PR is not intended to be merged.

File: packages/vats/test/bootstrapTests/supports.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/vats/test/bootstrapTests/supports.js:539]: I converted the PR to DRAFT to make it clear that it's not intended to be merged. The point of the PR is to show that an upgrade from current state to post-ZCF would work. It's a rough copy of #7966, which is intended to merge, but has to start from the post-ZCF world.

It doesn't seem worth spending effort to clean this up, as long as it's convincing. Let me know if you'd like me to do it anyway.

Comment (dckc) [reply packages/vats/test/bootstrapTests/supports.js:539]: ah. thanks for clarifying. sorry to bother you.

**Comment by Chris-Hibbert** [comment]
This branch has served its purpose. Now that #7900, #7946, #7966, and #8018 have been merged, it is no longer useful.

**Pull request closed without merging**